Userland/Libraries/LibJS/Runtime/StringConstructor.cpp

```diff
@@ -44,91 +44,158 @@ ThrowCompletionOr<void> StringConstructor::initialize(Realm& realm)
+    }
+    // 2. Else,
+    else {
+        // b. Let s be ? ToString(value).
+        primitive_string = TRY(value.to_primitive_string(vm));
+    }
+
+    // 4. Return StringCreate(s, ? GetPrototypeFromConstructor(NewTarget, "%String.prototype%")).
     auto* prototype = TRY(get_prototype_from_constructor(vm, new_target, &Intrinsics::string_prototype));
     return MUST_OR_THROW_OOM(StringObject::create(realm, *primitive_string, *prototype));
 }
 
-// 22.1.2.4 String.raw ( template, ...substitutions ), https://tc39.es/ecma262/#sec-string.raw
-JS_DEFINE_NATIVE_FUNCTION(StringConstructor::raw)
-{
-    auto* cooked = TRY(vm.argument(0).to_object(vm));
-    auto raw_value = TRY(cooked->get(vm.names.raw));
-    auto* raw = TRY(raw_value.to_object(vm));
-    auto literal_segments = TRY(length_of_array_like(vm, *raw));
-
-    if (literal_segments == 0)
-        return PrimitiveString::create(vm, String {});
-
-    auto const number_of_substituions = vm.argument_count() - 1;
-
-    StringBuilder builder;
-    for (size_t i = 0; i < literal_segments; ++i) {
-        auto next_key = DeprecatedString::number(i);
-        auto next_segment_value = TRY(raw->get(next_key));
-        auto next_segment = TRY(next_segment_value.to_deprecated_string(vm));
-
-        builder.append(next_segment);
-
-        if (i + 1 == literal_segments)
-            break;
-
-        if (i < number_of_substituions) {
-            auto next = vm.argument(i + 1);
-            auto next_sub = TRY(next.to_deprecated_string(vm));
-            builder.append(next_sub);
-        }
-    }
-    return PrimitiveString::create(vm, builder.to_deprecated_string());
-}
-
 // 22.1.2.1 String.fromCharCode ( ...codeUnits ), https://tc39.es/ecma262/#sec-string.fromcharcode
 JS_DEFINE_NATIVE_FUNCTION(StringConstructor::from_char_code)
 {
+    // 1. Let result be the empty String.
     Utf16Data string;
     string.ensure_capacity(vm.argument_count());
 
-    for (size_t i = 0; i < vm.argument_count(); ++i)
-        string.append(TRY(vm.argument(i).to_u16(vm)));
+    // 2. For each element next of codeUnits, do
+    for (size_t i = 0; i < vm.argument_count(); ++i) {
+        // a. Let nextCU be the code unit whose numeric value is ℝ(? ToUint16(next)).
+        auto next_code_unit = TRY(vm.argument(i).to_u16(vm));
 
+        // b. Set result to the string-concatenation of result and nextCU.
+        string.append(next_code_unit);
+    }
+
+    // 3. Return result.
     return PrimitiveString::create(vm, TRY(Utf16String::create(vm, move(string))));
 }
 
 // 22.1.2.2 String.fromCodePoint ( ...codePoints ), https://tc39.es/ecma262/#sec-string.fromcodepoint
 JS_DEFINE_NATIVE_FUNCTION(StringConstructor::from_code_point)
 {
+    // 1. Let result be the empty String.
     Utf16Data string;
-    string.ensure_capacity(vm.argument_count()); // This will be an under-estimate if any code point is > 0xffff.
+    // // This will be an under-estimate if any code point is > 0xffff.
```

Comment:
double `// //`

Reply:
Oops, only had to move this because clang-format wanted to right-align the next comment btw. Fixed!